### PR TITLE
DB Connection UI Docs

### PR DESCRIPTION
**Pull request by hughhhh**
Adding new documentation for creating new dynamic forms through for apache superset

### CODE REVIEW

File: src/pages/docs/Connecting to Databases/ui.mdx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+# in `superset/db_engine_specs/`.
+PREFERRED_DATABASES: List[str] = [
+    "PostgreSQL",
+    "Presto",
+    "MySQL",
+    "SQLite",
+]
+```
+
+For copyright reasons the logos for each database are not distributed with Superset (does anyone know if we can hotlink to them?).
```
Comment (eschutho) [inline src/pages/docs/Connecting to Databases/ui.mdx:40]: did we intend to leave the question in here?

File: src/pages/docs/Connecting to Databases/ui.mdx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+```python
+DB_IMAGES:
+  postgresql: "path/to/image/postgres.jpg"
+  bigquery: "path/to/s3bucket/bigquery.jpg"
+  snowflake: "path/to/image/snowflake.jpg"
+```
+
+### How to add new database engines to available endpoint
+
+Currently the new modal supports the following databases:
```
Comment (eschutho) [inline src/pages/docs/Connecting to Databases/ui.mdx:55]: do we plan to update this list as we add new dbs?

File: src/pages/docs/Connecting to Databases/ui.mdx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+- Postgres
+- Redshift
+- MySQL
+- BigQuery
+
+When the user selects a database not in this list they will see the old dialog asking for the SQLAlchemy URI. New databases can be added gradually to the new flow. In order to support the rich configuration a DB engine spec needs to have the following attributes:
+
+1. `parameters_schema`: a Marshmallow schema defining the parameters needed to configure the database. For Postgres this includes username, password, host, port, etc. ([see](https://github.com/apache/superset/blob/accee507c0819cd0d7bcfb5a3e1199bc81eeebf2/superset/db_engine_specs/base.py#L1309-L1320)).
+2. `default_driver`: the name of the recommended driver for the DB engine spec. Many SQLAlchemy dialects support multiple drivers, but usually one is the official recommendation. For Postgres we use "psycopg2".
```
Comment (eschutho) [inline src/pages/docs/Connecting to Databases/ui.mdx:65]: do we want to mention that if someone wants to use a different driver, they will need to use the sqlalchemy form?

Comment (eschutho) [reply src/pages/docs/Connecting to Databases/ui.mdx:65]: also maybe that the fields in the form map specifically to the driver itself, so if you load a different driver, then you won't be able to use the form.

File: src/pages/docs/Connecting to Databases/ui.mdx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+route: /docs/databases/db-connection-ui
+index: 1
+version: 1
+---
+
+**Add Giph for before and after**
+
+There are now 3 steps when connecting to a database:
+
+Step 1: First the admin must inform us what engine they want to connect to. This page is powered by the `/available` endpoint which pulls on the engines currently installed in your environment, so that only supported databases are shown.
```
Comment (eschutho) [inline src/pages/docs/Connecting to Databases/ui.mdx:15]: small nit.. maybe clarify "us"?

File: src/pages/docs/Connecting to Databases/ui.mdx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+name: Using Database Connection UI
+menu: Connecting to Databases
+route: /docs/databases/db-connection-ui
+index: 1
+version: 1
+---
+
+**Add Giph for before and after**
+
+There are now 3 steps when connecting to a database:
```
Comment (eschutho) [inline src/pages/docs/Connecting to Databases/ui.mdx]: another nit.. I'd recommend: `There are now 3 steps when connecting to a database in the new UI.`

File: src/pages/docs/Connecting to Databases/ui.mdx (hunk 3 above)
Comment (eschutho) [reply src/pages/docs/Connecting to Databases/ui.mdx:65]: nm, I see that you have that on line 60 👍

File: src/pages/docs/Connecting to Databases/ui.mdx (hunk 1 above)
Comment (hughhhh) [reply src/pages/docs/Connecting to Databases/ui.mdx:40]: yea we should remove that

File: src/pages/docs/Connecting to Databases/ui.mdx (hunk 2 above)
Comment (hughhhh) [reply src/pages/docs/Connecting to Databases/ui.mdx:55]: Yea i think we should so people know what dynamic forms currently avaialble

**Pull request closed without merging**